OpenRA.Game/Traits/World/ActorMap.cs

```diff
@@ -129,6 +133,7 @@ public void Tick(ActorMap am)
 				var delta = new WVec(range, range, WDist.Zero);
 				currentActors = am.ActorsInBox(position - delta, position + delta)
 					.Where(a => (a.CenterPosition - position).HorizontalLengthSquared < range.LengthSquared)
+					.Where(a => vRange.Length == 0 || (a.World.Map.DistanceAboveTerrain(a.CenterPosition).LengthSquared <= vRange.LengthSquared))
```

Comment:
For future reference, it would only walk the list once either way. It is marginally more efficient to combine the conditions though.